# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/gfx-hal/src/command/clear.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/gfx-hal/src/command/clear.rs b/third_party/rust/gfx-hal/src/command/clear.rs
deleted file mode 100644
--- a/third_party/rust/gfx-hal/src/command/clear.rs
+++ /dev/null
@@ -1,79 +0,0 @@
-use crate::pso;
-use std::fmt;
-
-/// A clear color union, which can be either `f32`, `i32`, or `u32`.
-#[repr(C)]
-#[derive(Clone, Copy)]
-pub union ClearColor {
-    /// `f32` variant
-    pub float32: [f32; 4],
-    /// `i32` variant
-    pub sint32: [i32; 4],
-    /// `u32` variant
-    pub uint32: [u32; 4],
-}
-
-impl fmt::Debug for ClearColor {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        writeln![f, "ClearColor"]
-    }
-}
-
-/// A combination of depth and stencil clear values.
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct ClearDepthStencil {
-    /// Depth value
-    pub depth: f32,
-    /// Stencil value
-    pub stencil: u32,
-}
-
-/// A set of clear values for a single attachment.
-///
-/// These are passed to a command buffer by
-/// [beginning a render pass][crate::command::CommandBuffer::begin_render_pass].
-#[repr(C)]
-#[derive(Clone, Copy)]
-pub union ClearValue {
-    /// Clear color
-    pub color: ClearColor,
-    /// Clear depth and stencil
-    pub depth_stencil: ClearDepthStencil,
-    _align: [u32; 4],
-}
-
-impl fmt::Debug for ClearValue {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.debug_struct("ClearValue")
-            .field("color", unsafe { &self.color.uint32 })
-            .field("depth_stencil", unsafe { &self.depth_stencil })
-            .finish()
-    }
-}
-
-impl Default for ClearValue {
-    fn default() -> Self {
-        ClearValue { _align: [0; 4] }
-    }
-}
-
-/// Attachment clear description for the current subpass.
-#[derive(Clone, Copy, Debug)]
-pub enum AttachmentClear {
-    /// Clear color attachment.
-    Color {
-        /// Index inside the `SubpassDesc::colors` array.
-        index: usize,
-        /// Value to clear with.
-        value: ClearColor,
-    },
-    /// Clear depth-stencil attachment.
-    DepthStencil {
-        /// Depth value to clear with.
-        depth: Option<pso::DepthValue>,
-        /// Stencil value to clear with.
-        stencil: Option<pso::StencilValue>,
-    },
-}